packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -0,0 +1,200 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import sanitize from 'sanitize-filename';
+import type NodeCache from 'node-cache';
+import type { ReaderCollection } from '@ui5/fs';
+import type { ToolsLogger } from '@sap-ux/logger';
+import type { NextFunction, Request, Response } from 'express';
+
+import { FolderNames, TemplateFileName, HttpStatusCodes } from '../types';
+
+interface WriteFragmentBody {
+    fragmentName: string;
+}
+
+/**
+ * @description Handles API Routes
+ */
+export default class RoutesHandler {
+    /**
+     * Constructor taking project as input.
+     *
+     * @param project Reference to the root of the project
+     * @param logger Logger instance
+     * @param cache Caching instance
+     */
+    constructor(
+        private readonly project: ReaderCollection,
+        private readonly logger: ToolsLogger,
+        private cache: NodeCache
+    ) {}
+
+    /**
+     * Generic caching function, caches the results of a callback
+     *
+     * @param key Cache key - unique identifier
+     * @param cb Callback used to return data
+     * @param ttlSeconds Time-to-live for cached item
+     * @template T
+     * @returns {T} Type or Interface of the return data
+     */
+    private withCache<T>(key: string, cb: () => T, ttlSeconds: number = 60): T {
```

Comment:
I don't think we should use caching here because
1. Do we really need that little performance benefit in a dev middleware?
2. How do you know that nobody manually added/removed a file since you filled the cache?